### PR TITLE
Feature Firebase Remote Config

### PR DESCRIPTION
**Pull request by AbdullahAhmedSoomro**
- Added Force Update Dialog.
- Implement Force Update for IOS and Android.

### CODE REVIEW

File: lib/presentation/service/remote_config.dart [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import 'package:firebase_remote_config/firebase_remote_config.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:package_info/package_info.dart';
+import 'package:star_book/presentation/shared/force_update_dialog.dart';
+
+class RemoteConfigKey {
+  static const String androidEnforcedVersion = 'androidEnforcedVersion';
+  static const String iosEnforcedVersion = 'iosEnforcedVersion';
+}
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:13]: could be an enum, with a string value

File: lib/presentation/service/remote_config.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+class ForceAppUpdate {
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:15]: Document how to setup on firebase

File: lib/presentation/service/remote_config.dart [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  static Future<void> enforcedVersion(BuildContext context) async {
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:16]: should work for production only. not for other envs

File: lib/presentation/service/remote_config.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    final remoteConfig = FirebaseRemoteConfig.instance;
+    await remoteConfig.setConfigSettings(
+      RemoteConfigSettings(
+        fetchTimeout: const Duration(minutes: 1),
+        minimumFetchInterval: const Duration(hours: 12),
+      ),
+    );
+    try {
+      final _key = Platform.isIOS
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:25]: prefer switch case over ternary

File: lib/presentation/service/remote_config.dart [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      final _enforcedVersion = remoteConfig.getString(_key);
+      final int currentVersion =
+          int.tryParse(_currentVersion.replaceAll('.', '')) ?? 0;
+      final int enforcedVersion =
+          int.tryParse(_enforcedVersion.replaceAll('.', '')) ?? 0;
+      if (enforcedVersion > currentVersion) {
+        if (context.mounted) {
+          await _forceAppUpdateDialogBox(context);
+        }
+        return;
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:40]: create a version class to encapsulate parsing and to compare. you may need operator overloading. this https://github.com/dartninja/version/blob/master/lib/version.dart may help

File: lib/presentation/service/remote_config.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      }
+    } catch (exception) {
+      log('Error on Remote Config: $exception');
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:43]: dont just log, need to report to firebase crashlytics

File: lib/presentation/service/remote_config.dart [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+  }
+
+  static Future<void> _forceAppUpdateDialogBox(context) async {
+    Future<bool> onBackButton() async {
+      SystemNavigator.pop();
```
Comment (hashirshoaeb) [inline lib/presentation/service/remote_config.dart:49]: that is new. i need more context.

File: pubspec.yaml [hunk 8]
```diff
@@ -37,6 +37,8 @@ dependencies:
   firebase_core: ^2.17.0
   firebase_crashlytics: ^3.3.7
   firebase_analytics: ^10.5.1
+  firebase_remote_config: ^4.3.2
+  package_info: ^2.0.2
```
Comment (hashirshoaeb) [inline pubspec.yaml:41]: why package_info ?

**Review by hashirshoaeb — APPROVED** [review]

File: pubspec.yaml (hunk 8 above)
Comment (AbdullahAhmedSoomro) [reply pubspec.yaml:41]: To get version of the app

Comment (hashirshoaeb) [reply pubspec.yaml:41]: we have package_info_plus already i believe.

**Pull request merged**